PoseEstimation-TFLiteSwift/VideoCapture.swift

```diff
@@ -0,0 +1,117 @@
+//
+//  VideoCapture.swift
+//  Awesome ML
+//
+//  Created by Eugene Bokhan on 3/13/18.
+//  Copyright © 2018 Eugene Bokhan. All rights reserved.
+//
+
+import UIKit
+import AVFoundation
+import CoreVideo
+
+public protocol VideoCaptureDelegate: class {
+    func videoCapture(_ capture: VideoCapture, didCaptureVideoFrame: CVPixelBuffer?, timestamp: CMTime)
+}
+
+public class VideoCapture: NSObject {
+    public var previewLayer: AVCaptureVideoPreviewLayer?
+    public weak var delegate: VideoCaptureDelegate?
+    public var fps = 30
+    
+    let captureSession = AVCaptureSession()
+    let videoOutput = AVCaptureVideoDataOutput()
+    let queue = DispatchQueue(label: "com.tucan9389.camera-queue")
+    
+    var lastTimestamp = CMTime()
+    
+    public func setUp(sessionPreset: AVCaptureSession.Preset = .vga640x480,
+                      cameraPosition: AVCaptureDevice.Position = .back,
+                      completion: @escaping (Bool) -> Void) {
+        self.setUpCamera(sessionPreset: sessionPreset,
+                         cameraPosition: cameraPosition,
+                         completion: { success in
+            completion(success)
+        })
+    }
+    
+    func setUpCamera(sessionPreset: AVCaptureSession.Preset, cameraPosition: AVCaptureDevice.Position, completion: @escaping (_ success: Bool) -> Void) {
+        
+        captureSession.beginConfiguration()
+        captureSession.sessionPreset = sessionPreset
+        
+        guard let captureDevice = AVCaptureDevice.default(.builtInWideAngleCamera,
+                                                          for: .video,
+                                                          position: cameraPosition) else {
+            
+            print("Error: no video devices available")
+            return
+        }
+        
+        guard let videoInput = try? AVCaptureDeviceInput(device: captureDevice) else {
+            print("Error: could not create AVCaptureDeviceInput")
+            return
+        }
+        
+        if captureSession.canAddInput(videoInput) {
+            captureSession.addInput(videoInput)
+        }
+        
+        let previewLayer = AVCaptureVideoPreviewLayer(session: captureSession)
```

Comment:
AVCaptureVideoPreviewLayer is a subclass of CALayer.
So, I recommend to replace this logic to View layer.